jctools-experimental/src/test/java/org/jctools/stacks/WFIStackTest.java

```diff
@@ -0,0 +1,395 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jctools.stacks;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import static org.junit.Assert.*;
+
+public class WFIStackTest
```

Comment:
We've been recently bitten by some unexpected concurrent behaviour of the JCTools data structures so I suggest to attempt writing some mpmc variation of tests although I understand the pain of making them reproducible/usefull

Reply:
What's a good example to follow for writing such a test?

Reply:
This one is a possible way to write it: https://github.com/JCTools/JCTools/blob/c0b570d31a4a60cb2b538f86106207f626c4b30c/jctools-core/src/test/java/org/jctools/queues/QueueSanityTest.java#L556

Using sequenced events to track expected order